src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -79,6 +79,40 @@ def customDID(pname):
     return pname
 
 
+def updateTransitionRecord(doc, userDN, logger):
+    """
+    Update transition record based on the following logic:
+    - check previous fraction value and add new transition record if it is
+    - differ from current one or does not exist
+      - first use-case: differ from container fraction of record
+      - second use-case: differ from last transition container fraction record
+    - https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5
+
+    :param doc: MSPileup record
+    :param userDN: user DN (string)
+    :param logger: logger to use
+    :return: nothing (document transition is updated in place)
+    """
+    fraction = doc.get('containerFraction', 1.0)
+    transition = doc.get('transition', [])
+    prevTranRecord = transition[-1]
+    if not userDN:
+        userDN = prevTranRecord.get('DN', '')
+        if not userDN:
+            msg = f'Last transition record {prevTranRecord} does not have DN attribute'
+            raise Exception(msg)
+    if prevTranRecord['containerFraction'] != fraction:
+        customName = customDID(prevTranRecord['customDID'])
+        # preserve previous container fraction
+        transitionRecord = {'containerFraction': prevTranRecord['containerFraction'],
```

Comment:
This line can potentially create another transition record with the same `containerFraction`. This is the second bug that we discussed today which is expected to be addressed with this PR.

Reply:
I intentionally didn't put it here since after thinking it through I think it is not relevant. Here is my logic:
- look at step 5 record content in [gist](https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5)
- over there we have two transition records with the same container fractions but both have different container fraction then fraction of the MSPileup record itself
- we used an artificial use-case when we provide via `updatePileupObjects .py` script a transition record while the code itself should never accept that since it is not designed for such use-case. Its logic when someone wants to update the document is to create transition record on its own.

Reply:
That's a good point! We cannot have that check against the previous to the last record indeed.

Bottom line (that we cannot ever forget):
* MSPileup REST call/action creates a transition record for us, with the containerFraction value before the actual update (e..g: containerFraction != transition/containerFraction). Only exception here is when a new pileup document is created, where containerFraction == transition/containerFraction.
* MSPileupTasks (daemon) does NOT create any transition, it simply updates the last one (containerFraction).